src/prefab_manager.zig

```diff
@@ -33,231 +29,52 @@ pub const PrefabManager = struct {
-                pbr_material.surface_type = .@"opaque";
-            }
-
-            if (material.pbr_metallic_roughness.base_color_texture.texture) |texture| {
-                if (texture.image) |image| {
-                    if (image.*.uri) |uri| {
-                        const texture_path = util.fromCStringToStringSlice(uri);
-                        const texture_path_u16 = @as([*:0]const u16, @ptrCast(&texture_path));
-                        pbr_material.albedo = gfxstate.scheduleLoadTexture(texture_path, .{ .state = d3d12.RESOURCE_STATES.COMMON, .name = texture_path_u16 }, arena) catch unreachable;
-                    }
-                }
-            }
-
-            if (material.pbr_metallic_roughness.metallic_roughness_texture.texture) |texture| {
-                if (texture.image) |image| {
-                    if (image.*.uri) |uri| {
-                        const texture_path = util.fromCStringToStringSlice(uri);
-                        const texture_path_u16 = @as([*:0]const u16, @ptrCast(&texture_path));
-                        pbr_material.arm = gfxstate.scheduleLoadTexture(texture_path, .{ .state = d3d12.RESOURCE_STATES.COMMON, .name = texture_path_u16 }, arena) catch unreachable;
-                    }
-                }
-            }
-
-            if (material.normal_texture.texture) |texture| {
-                if (texture.image) |image| {
-                    if (image.*.uri) |uri| {
-                        const texture_path = util.fromCStringToStringSlice(uri);
-                        const texture_path_u16 = @as([*:0]const u16, @ptrCast(&texture_path));
-                        pbr_material.normal = gfxstate.scheduleLoadTexture(texture_path, .{ .state = d3d12.RESOURCE_STATES.COMMON, .name = texture_path_u16 }, arena) catch unreachable;
-                    }
-                }
-            }
-
-            if (material.emissive_texture.texture) |texture| {
-                if (texture.image) |image| {
-                    if (image.*.uri) |uri| {
-                        const texture_path = util.fromCStringToStringSlice(uri);
-                        const texture_path_u16 = @as([*:0]const u16, @ptrCast(&texture_path));
-                        pbr_material.emissive = gfxstate.scheduleLoadTexture(texture_path, .{ .state = d3d12.RESOURCE_STATES.COMMON, .name = texture_path_u16 }, arena) catch unreachable;
-                    }
-                }
-            }
-
-            if (material.has_emissive_strength == 1) {
-                pbr_material.emissive_strength = material.emissive_strength.emissive_strength;
-            }
+    pub fn instantiatePrefab(self: @This(), world: ecsu.World, prefab: ecsu.Entity) ecsu.Entity {
+        const entity = world.newEntity();
+        entity.addPair(self.is_a, prefab);
+        return entity;
+    }
 
-            return gfxstate.storeMaterial(material_name, pbr_material) catch unreachable;
+    pub fn getPrefabByPath(self: *@This(), path: []const u8) ?ecsu.Entity {
```

Comment:
This looks like primarily a wrapper to get an idlocal from a string path. I think we should be able to put all of our IdLocals as "config" values (either computed at comptime or at init).

We can then have our APIs only work on IdLocals or the IdLocal hashes. Specifically I think we should avoid having IdLocals as the key in hashmaps (instead use the u64 hash), at least in release build, and in Debug only if the debugger can show us the key-value pair in a nice way.

Reply:
Nice! Is there a place where we do something like that in our codebase that I can look at?

Reply:
Yeah :)

For example `src\config\input.zig`

```
const ID = @import("../core/core.zig").ID;
pub const move_left = ID("move_left");
```

(unless I'm misunderstanding something)

Reply:
I'm probably missing some context on how this is used, e.g. if it comes from parsing a gltf string then we can't do it like that...  But I think I would prefer to make the conversion there rather than in the next step.. Soooo

Reply:
> (unless I'm misunderstanding something)

No, you didn't misunderstand. I hadn't used IdLocals in a while so I needed a quick refresher.

> I'm probably missing some context on how this is used, e.g. if it comes from parsing a gltf string then we can't do it like that... But I think I would prefer to make the conversion there rather than in the next step.. Soooo

At the moment it's simply hardcoded in the code in 2 places
1. where we load the prefabs
2. where we use the prefabs

There's no parsing of source assets names or anything comptime.

Should I declare these IDs in `src\config\prefabs.zig`, right?

Reply:
instead of:
```
const char* lol = getFromJson("name")
const prefab = getPrefabByPath(lol)
```

I would prefer:
```
const lolid = getIDFromJson("name")
const prefab = getPrefab( lolid);
```

rationale being that it would (in my mind) be less encouraging to use strings more than necessary and instead make it clear that we use Idlocal and hashes as much as possible internally.

Reply:
> Should I declare these IDs in `src\config\prefabs.zig`, right?

Yep sounds good!

Reply:
I'm not sure I'm getting what that `getIDFromJson` function should do. What I understood was something like this:

```
pub const player_id = ID("prefabs/characters/player/player.bin");
```